src/1.JWAS/src/Nonlinear/nonlinear.jl

```diff
@@ -7,19 +7,49 @@
 # (2) neural network: a1*tan(x1)+a2*tan(x2)
 
 #nonlinear_function: #user-provide function, "tanh"
+
+#ycorr: residual for omics data
 function sample_latent_traits(yobs,mme,ycorr,nonlinear_function)
     ylats_old = mme.ySparse         # current values of each latent trait; [trait_1_obs;trait_2_obs;...]
     μ_ylats   = mme.ySparse - ycorr # mean of each latent trait, [trait_1_obs-residuals;trait_2_obs-residuals;...]
                                     # = vcat(getEBV(mme,1).+mme.sol[1],getEBV(mme,2).+mme.sol[2]))
-    σ2_yobs   = mme.σ2_yobs         # residual variance of yobs (scalar)
+    σ2_yobs      = mme.user_σ2_yobs==false ? mme.σ2_yobs : mme.user_σ2_yobs    # residual variance of yobs (scalar)
+    σ2_weightsNN = mme.user_σ2_weightsNN==false ? mme.σ2_weightsNN : mme.user_σ2_weightsNN
```

Comment:
why having mme.user_σ2_yobs and mme.user_σ2_weightsNN? Why not just use mme.σ2_yobs and mme.σ2_weightsNN. Add an indicator for fixed or not fixed?